docs/developer-guide/udf.rst

```diff
@@ -104,6 +104,21 @@ The KSQL server will check the value being passed to each parameter and report a
 log for any null values being passed to a primitive type. The associated column in the output row
 will be ``null``.
 
+
+Dynamic return type
+~~~~~~~~~~~~~~~~~~~
+
+UDFs support dynamic return types that are resolved at runtime. This is for example useful if you
+want to implement a UDF whose argument and return type are ``BigDecimal`` and you want to define the
```

Comment:
```suggestion
want to implement a UDF that has an argument and return type of ``BigDecimal`` and you want to define the
```

Reply:
@JimGalasyn  -  I think this suggestion loses some of the meaning of the doc. @vpapavas I think maybe we shoudl write "This is useful if you want to implement a UDF with a non-deterministic return type. A UDF which returns  \`\`BigDecimal\`\`, for example, may vary the precision and scale of the output based on the input schema.